### DISCUSSION TITLE
only_light/only_dark classes break alignment of images within figures

### DISCUSSION THREAD

**Discussion by idryzhov** [pradyunsg/furo#423] [Potential Issue] [answered]
When I use `:class: only_light` or `:class: only_dark` on a figure, the image within the figure is left-aligned although it should be centered.
As I understand, this happens because the `display: block !important` property is added to such images. The property was removed in https://github.com/pradyunsg/furo/commit/b414d2fe3e281a52d3d2e1e59efebc155d6c06b3 and later was brought back in https://github.com/pradyunsg/furo/commit/3bcbe2ce404080061481638efff8e6102b3b812d. There's no comment in the linked commits about the necessity of this property, so my question here – why do we need this property and is there any other property that we can use there that won't break the alignment?

**Comment by pradyunsg** [comment]
Hi there! Could you provide a reproducer for the issue you're seeing?

**Reply by idryzhov** [reply]
Hi! Thanks for a quick reply.

It's pretty simple. When I use:
```
.. figure:: https://source.unsplash.com/200x200/daily?cute+dogs
    :class: only_light

.. figure:: https://source.unsplash.com/200x200/daily?cute+cats
    :class: only_dark
```
The image is left-aligned:
<img width="796" alt="Screenshot 2022-04-17 at 20 02 46" src="https://user-images.githubusercontent.com/1313087/163724848-a0350465-8971-4c9f-8e17-f9ceb2740c2a.png">
However, I expect it to be centered, as with regular `figure`, when I don't use `only-light`/`only-dark` classes:
<img width="794" alt="Screenshot 2022-04-17 at 20 04 24" src="https://user-images.githubusercontent.com/1313087/163724903-ea50b610-1b75-4a74-a865-631e452c9610.png">

**Comment by idryzhov** [comment, marked as answer]
@pradyunsg hi again!

Actually, I think I found the right solution.
When using figures, one must use `:figclass: only_light/dark` instead of `:class: only_light/dark`.
This way, the class applies to the `<figure>` HTML tag instead of the `<img>` and the image is aligned correctly.
More than that, it allows to use different captions depending on the light/dark mode:
```
.. figure:: https://source.unsplash.com/200x200/daily?cute+dogs
    :figclass: only_light
    :align: center

    Dog

.. figure:: https://source.unsplash.com/200x200/daily?cute+cats
    :figclass: only_dark
    :align: center

    Cat
```

So there's no issue with the Furo theme, sorry. However, the usage of `:figclass: only_light/dark` for figures may be worth noting in the docs :)

**Reply by pradyunsg** [reply]
I've gone ahead and documented this as well.
